In [1]:
import pandas
import matplotlib.pyplot as plt
import math
import numpy as np
from scipy.stats import hypergeom
from scipy.stats import binom
from scipy import stats as st
import scipy.special


 **Aufgabe 3 a)**

**KENO-Gewinnquote bei 10 getippten Zahlen**

In [2]:
einsatz = 2
exp_val = 0
std_var = 0
var = 0

rv_10 = st.hypergeom(70, 20, 10)
x_10 = (10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
profit_10 = (100000, 1000, 100, 15, 5, 2, 0, 0, 0, 0, 2)
#profit_10 = (199998, 1998, 198, 28, 8, 2,-2, -2, -2, -2, 2)

for i, x in enumerate(x_10):
    exp_val += (((profit_10[i]) * einsatz - 2)* rv_10.pmf(x))
    
for i, x in enumerate(x_10):
    var += (rv_10.pmf(x) * ((einsatz*profit_10[i] -2)-exp_val)**2)
    
std_var = math.sqrt(var)

In [3]:
print("X10:")
print(f"Erwartungswert: {exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X10:
Erwartungswert: -1.012005
Varianz: 18735.7766
Standardabweichung: 136.8787


**Aufgabe 3 b)**

**KENO-Gewinnquote bei 9 getippten Zahlen**

In [4]:
exp_val = 0
std_var = 0
var = 0

rv_9 = st.hypergeom(70, 9, 20)
x_9 = (9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
profit_9 = (50000, 1000, 20, 5, 2, 0, 0, 0, 0, 2)

for i, x in enumerate(x_9):
    exp_val += (((profit_9[i]) * einsatz - 2)* rv_9.pmf(x))
    
for i, x in enumerate(x_9):
    var += (rv_9.pmf(x) * ((einsatz*profit_9[i] -2)-exp_val)**2)
    
std_var = math.sqrt(var)

In [5]:
print("X9:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X9:
Erwartungswert:-0.999094
Varianz: 26218.0757
Standardabweichung: 161.9200


**KENO-Gewinnquote bei 8 getippten Zahlen**

In [6]:
exp_val = 0
std_var = 0
var = 0

rv_8 = st.hypergeom(70, 20, 8)
x_8 = (8, 7, 6, 5, 4, 3, 2, 1, 0)
profit_8 = (10000, 100, 15, 2, 1, 0, 0, 0, 1)

for i, x in enumerate(x_8):
    exp_val += (((profit_8[i]) * einsatz - 2)* rv_8.pmf(x))
    
for i, x in enumerate(x_8):
    var += (rv_8.pmf(x) * ((einsatz*profit_8[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [7]:
print("X8:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X8:
Erwartungswert:-1.021233
Varianz: 5358.7205
Standardabweichung: 73.2033


**KENO-Gewinnquote bei 7 getippten Zahlen**

In [8]:
exp_val = 0
std_var = 0
var = 0

rv_7 = st.hypergeom(70, 20, 7)
x_7 = (7, 6, 5, 4, 3, 2, 1, 0)
profit_7 = (1000, 100, 12, 1, 0, 0, 0, 0)

for i, x in enumerate(x_7):
    exp_val += (((profit_7[i]) * einsatz - 2)* rv_7.pmf(x))
    
for i, x in enumerate(x_7):
    var += (rv_7.pmf(x) * ((einsatz*profit_7[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [9]:
print("X7:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X7:
Erwartungswert:-1.008670
Varianz: 331.7899
Standardabweichung: 18.2151


**KENO-Gewinnquote bei 6 getippten Zahlen**

In [10]:
exp_val = 0
std_var = 0
var = 0

rv_6 = st.hypergeom(70, 20, 6)
x_6 = (6, 5, 4, 3, 2, 1, 0)
profit_6 = (500, 15, 2, 1, 0, 0, 0, 0)

for i, x in enumerate(x_6):
    exp_val += (((profit_6[i]) * einsatz - 2)* rv_6.pmf(x))
    
for i, x in enumerate(x_6):
    var += (rv_6.pmf(x) * ((einsatz*profit_6[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [11]:
print("X6:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X6:
Erwartungswert:-1.005121
Varianz: 301.3534
Standardabweichung: 17.3595


**KENO-Gewinnquote bei 5 getippten Zahlen**

In [12]:
exp_val = 0
std_var = 0
var = 0

rv_5 = st.hypergeom(70, 20, 5)
x_5 = (5, 4, 3, 2, 1, 0)
profit_5 = (100, 7, 2, 0, 0, 0)

for i, x in enumerate(x_5):
    exp_val += (((profit_5[i]) * einsatz - 2)* rv_5.pmf(x))
    
for i, x in enumerate(x_5):
    var += (rv_5.pmf(x) * ((einsatz*profit_5[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [13]:
print("X5:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X5:
Erwartungswert:-1.002042
Varianz: 56.0134
Standardabweichung: 7.4842


**KENO-Gewinnquote bei 4 getippten Zahlen**

In [14]:
exp_val = 0
std_var = 0
var = 0

rv_4 = st.hypergeom(70, 20, 4)
x_4 = (4, 3, 2, 1, 0)
profit_4 = (22, 2, 1, 0, 0)

for i, x in enumerate(x_4):
    exp_val += (((profit_4[i]) * einsatz - 2)* rv_4.pmf(x))
    
for i, x in enumerate(x_4):
    var += (rv_4.pmf(x) * ((einsatz*profit_4[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [15]:
print("X4:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X4:
Erwartungswert:-1.011141
Varianz: 11.2623
Standardabweichung: 3.3559


**KENO-Gewinnquote bei 3 getippten Zahlen**

In [16]:
exp_val = 0
std_var = 0
var = 0

rv_3 = st.hypergeom(70, 20, 3)
x_3 = (3, 2, 1, 0)
profit_3 = (16, 1, 0, 0)

for i, x in enumerate(x_3):
    exp_val += (((profit_3[i]) * einsatz - 2)* rv_3.pmf(x))
    
for i, x in enumerate(x_3):
    var += (rv_3.pmf(x) * ((einsatz*profit_3[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [17]:
print("X3:")
print(f"Erwartungswert:{exp_val:.6f}")
print(f"Varianz: {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X3:
Erwartungswert:-0.986482
Varianz: 20.9925
Standardabweichung: 4.5818


**KENO-Gewinnquote bei 2 getippten Zahlen**

In [18]:
exp_val = 0
std_var = 0
var = 0

rv_2 = st.hypergeom(70, 20, 2)
x_2 = (2, 1, 0)
profit_2 = (6, 0, 0)

for i, x in enumerate(x_2):
    exp_val += (((profit_2[i]) * einsatz - 2)* rv_2.pmf(x))
    
for i, x in enumerate(x_2):
    var += (rv_2.pmf(x) * ((einsatz*profit_2[i] -2)-exp_val)**2)

std_var = math.sqrt(var)

In [19]:
print("X2:")
print(f"Erwartungswert: {exp_val:.6f}")
print(f"Varianz {var:.4f}")
print(f"Standardabweichung: {std_var:.4f}")

X2:
Erwartungswert: -1.055901
Varianz 10.4379
Standardabweichung: 3.2308


**Aufgabe 3 c)**

Die Bank gewinnt immer!
Der Spieler ist immer im nachteil und wird langfristig immer öfter Verlust erziehlen als gewinn, da der Erwartungswert immer unter 0 ist.
Aber mit 3 Kästchen hat der Spieler bei Keno3 die höchste Chance zu gewinnen mit einem Erwartungswert von -0.986482.

**Aufgabe 3 d)**

In [20]:
rv_h = st.hypergeom(70, 20, 6)

ereignis = rv_h.pmf(0) + rv_h.pmf(1) + rv_h.pmf(2)

rv_b = binom(4,ereignis)

print('Wahrscheinlichkeit: p = %1.4f' %rv_b.pmf(3))


Wahrscheinlichkeit: p = 0.4181


Den Verlust bei 0, 1 oder 2 Richtigen habe ich mit mit der Binomialverteilung berechnet.
Bei der Ausgabe habe ich .pfm weil das Ergebniss genau auf 3-mal sein muss.